### PR TITLE
Add support and switch between of rsa2048, rsa4096 and ed25519 keys

### PR DESCRIPTION
**Pull request by patriotyk**
Running all uptane related tests with command `ctest -R test_uptane` gives me the following results:
```
RSA2048 : 120 secs
ED255519: 60 secs
```
So looks like ED25519 is much faster then RSA2048. 
But server doesnt support it, so we can not use it by default yet.

### CODE REVIEW

File: src/crypto.cc [hunk 1]
```diff
@@ -95,6 +95,22 @@ std::string Crypto::RSAPSSSign(ENGINE *engine, const std::string &private_key, c
   return retval;
 }
 
+std::string Crypto::Sign(KeyType key_type, ENGINE *engine, const std::string &private_key, const std::string &message) {
+  if (key_type == kED25519) {
+    std::cout << "PRIVKEY:" << private_key;
```
Comment (lbonn) [inline src/crypto.cc]: Forgot a debugging print here

**Comment by cajun-rat** [comment]
Let's hold off merging this until the Debian stuff is done and we have a working release of that.  I don't want to merge anything risky until we have this out the door.

File: src/uptane/secondaryconfig.h [hunk 2]
```diff
@@ -22,6 +23,7 @@ class SecondaryConfig {
   bool partial_verifying;
   std::string ecu_private_key;
   std::string ecu_public_key;
+  KeyType key_type{kRSA2048};
```
Comment (patrickvacek) [inline src/uptane/secondaryconfig.h]: This is basically hardcoding all secondaries to use RSA 2048. That's not terrible, but we should maybe instead add the ability to set this via `Config::readSecondaryConfigs` and change our examples and test data to try it out.

File: src/config.h [hunk 3]
```diff
@@ -79,7 +80,8 @@ struct UptaneConfig {
         primary_ecu_hardware_id(""),
         director_server(""),
         repo_server(""),
-        key_source(kFile) {}
+        key_source(kFile),
+        key_type(kRSA2048) {}
```
Comment (patrickvacek) [inline src/config.h]: I think we need to be able to set this via the config file. It needs to be handled in `Config::updateFromPropertyTree` and `Config::writeToFile`.

**Comment by patriotyk** [comment]
Looks like there is no any benefits in speedin CI. So I will update PR with default ED keys for all tests to check if something change

**Comment by patriotyk** [comment]
I think it is ok now. However there is no any speedup profit comparing to RSA2048 even when run all tests with ED25519. But on my local machine ED25519 performs much faster

**Comment by cajun-rat** [comment]
Benchmark: 

## With these changes (3 runs)

~~~
ninja clean && time ninja qa

Total Test time (real) = 189.70 sec
real	4m26,985s
user	15m30,724s
sys	0m33,272s

Total Test time (real) = 170.85 sec
real	4m8,790s
user	15m10,832s
sys	0m34,140s

Total Test time (real) = 185.23 sec
real	4m23,467s
user	15m23,628s
sys	0m34,220s

~~~

## without (master): 

~~~
Total Test time (real) = 191.73 sec
real	4m29,087s
user	15m26,884s
sys	0m33,652s

Total Test time (real) = 198.52 sec
real	4m36,718s
user	15m38,312s
sys	0m34,048s

Total Test time (real) = 207.30 sec
real	4m45,438s
user	15m34,548s
sys	0m34,204s
~~~

**Pull request merged**